### DISCUSSION TITLE
Draft paper on the use of APIs and automated updates in HEOR

### DISCUSSION THREAD

**Discussion by RobertASmithBresMed** [RobertASmithBresMed/plumberHE#22] [Announcements]
It would be great to get feedback on the draft paper within this repo:

https://github.com/RobertASmithBresMed/plumberHE/blob/main/report/academicpaper.pdf

**Comment by RobertASmithBresMed** [comment]
- I feel that the generation of a pdf when the results are back with the consultant distracts from the message of your paper. Feels more central than it is. As it is just an example of what could be the next step. I suggest making it explicit throughout the paper I.e. "Once the results are back with the consultant using this platform, it can be used in downstream tasks such as rendering a pdf with further analysis).

- My first question, without reading the rest of the paper is: how would you ensure that what comes out of your model doesn't include sensitive data. I.e. if the consultant has a way to execute code remotely into the company's server, and getting the result without intervention, nothing stops malicious attempts to gain sensitive data. Unless the output is revised against by someone/something. For example, attempting to automatically match sensitive inputs in the outputs, could be an efficient way to help mitigate the leaking of data. (however the data could also be cyphered before being sent back, then this would not work).

**Comment by ghost** [comment]
Great paper, I think moving into the right direction. There are many outlined benefits when the data is held secure in place, and only the algorithm is what exchanges hands. First one is the reduction of possible data leaks, but also less evident ones such as the interoperability of systems that APIs provide. 

My only concern again, is data leaks. I believe that a malicious user, could include code to obsfuscate the sensitive data before the output is returned by the API. Which shows the need for someone within the company to review the code being executed through the API.

However, the benefits of using this approach can be a good middle point between giving consultants direct access to sensitive data, and the highly restrictive safe-havens, which can hinder the development process.

In fact this solution can help safe-havens operate lighter, if the dedicated personel only have to review the code being executed, ensuring the are no intended data leaks. Rather than having to review the data coming both in and out, which is much more labour intensive, and potentially leading to more human-related errors.
